src/views/List.jsx

```diff
@@ -1,16 +1,47 @@
 import { ListItem } from '../components';
+import { useState } from 'react';
 
 /** List component that displays items in a user's shopping cart  */
 export function List({ data }) {
+	const [filteredList, setFilteredList] = useState([]);
+	const [filterInput, setFilterInput] = useState('');
+
+	/* Use handler to change the state of filterInput  and convert all items to lowercase to facilitate a more thorough search */
+	const handleInput = (event) => {
+		const value = event.target.value.toLowerCase();
+		setFilterInput(value);
+		setFilteredList(
+			data.filter((item) => {
+				return item.name && item.name.toLowerCase().includes(value);
+			}),
+		);
+	};
+
 	return (
 		<>
 			<p>
 				Hello from the <code>/list</code> page!
 			</p>
+
+			<form>
+				<label htmlFor="list-filter">Filter items</label>
+				<br />
+				<input
+					id="list-filter"
+					type="search"
+					placeholder="Start typing here..."
+					onChange={handleInput}
+				/>
+			</form>
+			{/* Uses data or state of filteredList depending on state of filterInput */}
 			<ul>
-				{data.map((item) => {
-					return <ListItem key={item.id} name={item.name} />;
-				})}
+				{!filterInput
```

Comment:
Definitely not a blocking comment, but for readability I prefer to move this kind of logic out into a separate function outside of the component's return statement. This is fine for now, but as more is added here, this can in my opinion make the JSX and relevant operations difficult to digest.

So it could look something like this:
`const renderItems = () => { ...add in what you want to do here... }`

Then in the return statement, you can just add `{renderItems()}`

Reply:
@djtaylor8 I tried this a few different ways and couldn't seem to make it work. I'm adding a screenshot of one of the things I tried, and I don't get any errors, but none of the lists render. Thoughts on what may be wrong??
![Screen Shot 2023-01-27 at 5 57 34 PM](https://user-images.githubusercontent.com/107324271/215232464-2fb9e786-08fc-4a2f-8f94-74b82f4f3a2d.png)
![Screen Shot 2023-01-27 at 5 57 59 PM](https://user-images.githubusercontent.com/107324271/215232493-8db0c2d0-37b6-4f2a-a6dc-550430e69730.png)

Reply:
@sarahmjenkins Looks like adding `return` just before `!filterInput` works.

Reply:
🤦🏻‍♀️ of course!!! Thank you!! My brain was clearly running out of steam!

Reply:
Story of my life @sarahmjenkins! Great catch, @arcan9!